docs/backend-specialist/2.3-entity-api.md

```diff
@@ -0,0 +1,485 @@
+ *       "default" = "Drupal\config_pages\ConfigPagesForm"
+ *     },
+ *     "translation" = "Drupal\config_pages\ConfigPagesTranslationHandler"
+ *   },
+ *   admin_permission = "administer config_pages types",
+ *   base_table = "config_pages",
+ *   links = {
+ *     "canonical" = "/config_pages/{config_pages}",
+ *     "edit-form" = "/config_pages/{config_pages}",
+ *     "collection" = "/admin/structure/config_pages/config-pages-content",
+ *   },
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "bundle" = "type",
+ *     "label" = "label",
+ *     "context" = "context",
+ *     "uuid" = "uuid"
+ *   },
+ *   bundle_entity_type = "config_pages_type",
+ *   field_ui_base_route = "entity.config_pages_type.edit_form",
+ *   render_cache = TRUE,
+ * )
+ */
+class ConfigPages extends ContentEntityBase implements ConfigPagesInterface {
+```
+You can also add RouteProvider, form handlers, operations, etc.
+
+### Base field definitions
+This is important part of content entity implementation. Base fields are the ones that will be a columns in main entity data DB table. You can specify type of the field, title, default value, and even field widget and formatter (but you don't have to if not needed).
+Implement method like this:
+```php
+/**
+   * {@inheritdoc}
+   */
+  public static function baseFieldDefinitions(EntityTypeInterface $entity_type): array {
+
+    $fields = parent::baseFieldDefinitions($entity_type);
+    
+    $fields['name'] = BaseFieldDefinition::create('string')
+      ->setLabel(t('Name'))
+      ->setDescription(t('The name of the test entity.'))
+      ->setTranslatable(TRUE)
+      ->setSetting('max_length', 32)
+      ->setDisplayOptions('view', [
+        'label' => 'hidden',
+        'type' => 'string',
+        'weight' => -5,
+      ])
+      ->setDisplayConfigurable('form', TRUE) // to make field widget for edit in the UI
+      ->setDisplayOptions('form', [
+        'type' => 'string_textfield',
+        'weight' => -5,
+      ]);
+
+    $fields['shared_by'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Shared by'))
+      ->setDescription(t('The ID of the sharer.'))
+      ->setSetting('target_type', 'user');
+      // ->isReadOnly(TRUE) will make field readonly.
+
+    $fields['shared_to'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Shared to'))
+      ->setDescription(t('The ID of the receiver.'))
+      ->setSetting('target_type', 'user');
+
+    $fields['created'] = BaseFieldDefinition::create('created')
+      ->setLabel(t('Shared on'))
+      ->setDescription(t('The time that the favorites list was shared.'));
+
+    return $fields;
+  }
+```
+
+More info on possible configurations in annotation: https://www.drupal.org/docs/drupal-apis/entity-api/introduction-to-entity-api-in-drupal-8
+
+## Bundles
+
+Sometimes, a content entity type needs bundles. For example, a **Node** is a content entity, while its bundles are **NodeType** config entities.
+
```

Comment:
Want to add something like this?
A content entity type can have multiple bundles, which are essentially subtypes of the entity. For example, the Node entity type has bundles like Article and Page.  
So:
* Content Entity Type: A general category of content, such as Node, User, or Taxonomy Term.
* Bundle: A specific subtype of a content entity type, such as Article or Page for the Node entity type.

Reply:
Added